modules/tensor_mechanics/src/actions/DomainIntegralAction.C

```diff
@@ -136,6 +138,12 @@ DomainIntegralAction::DomainIntegralAction(const InputParameters & params)
   {
     _crack_front_points = getParam<std::vector<Point>>("crack_front_points");
   }
+  if (isParamValid("crack_front_points_provider") && isParamValid("number_crack_front_points"))
+  {
+    _use_crack_front_points_provider = true;
+    _crack_front_points_provider = getParam<UserObjectName>("crack_front_points_provider");
+    _number_crack_front_points = getParam<unsigned int>("number_crack_front_points");
+  }
```

Comment:
I think you need an error check for the case where 'crack_front_points_provider' is supplied, but 'number_crack_front_points' isn't.

Reply:
Actually, maybe that's covered by your error check in CrackFrontDefinition.